src/rules/function-url-scheme-whitelist/index.js

```diff
@@ -0,0 +1,57 @@
+import {
+  functionArgumentsSearch,
+  isStandardSyntaxUrl,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+import { isString, trim } from "lodash"
+import { parse } from "url"
+
+export const ruleName = "function-url-scheme-whitelist"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (scheme) => `Unexpected url scheme ${scheme}:`,
+})
+
+export default function (whitelist) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: whitelist,
+      possible: [isString],
+    })
+    if (!validOptions) { return }
+
+    root.walkDecls(function (decl) {
+      functionArgumentsSearch(decl.toString().toLowerCase(), "url", (args, index) => {
+        const urlString = trim(args, " '\"")
+        if (!isStandardSyntaxUrl(urlString)) { return }
+
+        const url = parse(urlString)
+        if (url.protocol === null) { return }
+
+        // The URL spec does not require a scheme to be followed by `//`, but checking
+        // for it allows this rule to differentiate <scheme>:<hostname> urls from
+        // <hostname>:<port> urls. `data:` scheme urls are an exception to this rule.
+        const slashIndex = url.protocol.length
+        const expectedSlashes = urlString.slice(slashIndex, slashIndex + 2)
+        const isSchemeLessUrl = (
+          expectedSlashes !== "//" &&
+          url.protocol !== "data:"
+        )
+        if (isSchemeLessUrl) { return }
+
+        const scheme = url.protocol.slice(0, -1) // strip trailing `:`
+        if (whitelist.includes(scheme)) { return }
```

Comment:
> I almost used it, but had a hard time thinking of a situation where it'd be useful for this rule.

Yes you're right. I guess the same can be said for `unit-whitelist` and `unit-blacklist`. 

Shall we just leave it as is?